### DISCUSSION TITLE
How to show updated date in blog post?

### DISCUSSION THREAD

**Discussion by nicfv** [squidfunk/mkdocs-material#6595] [Q&A] [answered]
What do I need to do to enable what's at the bottom of this screenshot?

![image](https://github.com/squidfunk/mkdocs-material/assets/24358775/a370756d-90c3-42b0-8c52-b1fa0f86799a)

Thanks!

**Comment by kamilkrzyskow** [comment]
Hello @nicfv,
everything is explained here: https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#revisioning
You need to install the `mkdocs-git-revision-date-localized-plugin` Python plugin, and add it `mkdocs.yml`.
The theme then uses the information provided by the plugin and then renders the info when building the page:
https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/source-file.html

**Reply by nicfv** [reply]
Thanks for getting back to me so quickly! I'll definitely try out the revision date localized plugin. However, is there not a way to simply display the last updated date from the page's metadata? For example,

```md
---
date: 2023-12-14
date_updated: 2024-01-04 # this one
categories:
  - Example
---

# Page title

Words...
```

**Reply by kamilkrzyskow** [reply]
> is there not a way to simply display the last updated date from the page's metadata?

An already built-in solution does not exist according to my knowledge. Also I guess you probably want it to be displayed in the format from the screenshot like `1 month ago`. So the `date_updated` would need to be converted into a relative time format. 

However, you could override the `partials/source-file.html` yourself and implement a hook that exposes a date converter to the Jinja2 environment. So in the partial you'd access your custom `date_updated` value and use that exposed date converter.
https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme
https://www.mkdocs.org/user-guide/configuration/#hooks

🤔💡 Having written that, perhaps there is a time converter already available from the Blog plugin, so there wouldn't be a need for your custom hook. I shall investigate. 
Nonetheless, modifying the partial is required, so I wouldn't call it simple and straightforward, as for example installing a plugin and enabling it in `mkdocs.yml`

**Reply by nicfv** [reply]
The relative format is nice but not necessary. The metadata shown on the left bar of the page would also be a great place to show the last updated date.

![image](https://github.com/squidfunk/mkdocs-material/assets/24358775/6748494e-89b9-4f30-b4d3-393d9df70a71)

I checked in [blog-post.html](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/blog-post.html) and it appears that there actually is code to render the updated date.

```html
<!-- Post date updated -->
{% if page.config.date.updated %}
  <li class="md-nav__item">
    <div class="md-nav__link">
      {% include ".icons/material/calendar-clock.svg" %}
      <time
        datetime="{{ page.config.date.updated }}"
        class="md-ellipsis"
      >
        {{- page.config.date.updated | date -}}
      </time>
    </div>
  </li>
{% endif %}
```

So I did some more digging. From looking up "updated" in the docs website, I found this, which matches the path `page.config.date.updated` from above:

![image](https://github.com/squidfunk/mkdocs-material/assets/24358775/b65e1997-6830-464c-9cba-6a06f19ebe94)

(Try actually visiting [writing your first post](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/?h=updated#writing-your-first-post). That example actually isn't there, but that's a different issue.)

So I tried changing my page metadata to the following, ~~but I still don't get the updated date. Any ideas?~~

```md
---
date:
  created: 2021-05-08
  updated: 2022-05-09
...
---
```

Actually, I just needed to wait a few minutes for it to propagate on my GH pages... That's all I really needed to do, I guess! I just had the wrong metadata format for the date updated.

**Reply by kamilkrzyskow** [reply]
After some investigation I can say that unfortunately, the blog plugin doesn't expose a date converter that would meet your needs.
It exposes a `date` filter that converts the date in the format seen in the side panel of a blog post.
https://github.com/squidfunk/mkdocs-material/blob/246266fafad2c2fdc78022c14f7f4b2df91167f3/src/plugins/blog/plugin.py#L300-L326
https://github.com/squidfunk/mkdocs-material/blob/246266fafad2c2fdc78022c14f7f4b2df91167f3/src/templates/blog-post.html#L88-L99

So you can mimic this to create your own hook that exposes a custom `date` filter which converts your `date_updated` into a format like `1 month ago`.

**Reply by nicfv** [reply, marked as answer]
Thanks for checking this out. My main issue was just that I had `date_updated` instead of `date.updated`. It would be interesting to have a builtin relative date converter. For now I'll just try out the plugin you suggested.

**Reply by kamilkrzyskow** [reply]
Oh, nice find @nicfv. I didn't find the `updated` since I mostly looked at the code side of things. 
So if you found what you wanted then mark your answer as your answer 😉, as I hit a wall with this one.

**Comment by jaywhj** [comment]
@nicfv 

This new plugin solves your problem perfectly [mkdocs-document-dates](https://github.com/jaywhj/mkdocs-document-dates), try it!

![render](https://github.com/user-attachments/assets/56fe9586-9af1-4de9-86ab-9c0c3d1d63a1)
